Ghidra/Processors/SuperH/data/languages/superh.sinc

```diff
@@ -929,6 +1329,26 @@ target00_11: target is sdisp_00_11 [ target = (sdisp_00_11 << 1) + inst_start +
     rn_08_11 = (rn_08_11 >> 1) | sext($(T_FLAG));
 }
 
+:shad rm_04_07, rn_08_11   is opcode_12_15=0b0100 & rn_08_11 & rm_04_07 & opcode_00_03=0b1100
+{
+    local sgn = rm_04_07 & 0x80000000;
```

Comment:
I see another potential issue here. According to the processor spec, shad is an SH-2A instruction but not an SH-1 or SH-2 instruction. It appears that this instruction is not #ifdef-ed properly and is currently enabled for SH-1/SH-2/SH-2A when it should only be enabled for SH-2A. I will look further into this. 

I honestly think it might be cleaner to wrap every instruction in an #ifdef SH1|SH2|SH2A. We will avoid any issues like this and make it easier for future SuperH contributions. But it will be annoying to do.

Reply:
Also, it's possible to split into files where SH-1 is the base and each new core adds the extra instructions in a separate file, or maybe splitting to files by instruction type, but then you don't lose the ifdefs...

Note that some instructions, such as `sts` for example were modified between SH-1 and SH-2, I haven't encountered such instructions when adding SH-2A though. So there are more "internal" `ifdefs`.

<pre>
:sts    mach,rn_08_11  is mach & opcode_12_15=0b0000 & rn_08_11 & opcode_00_07=0b00001010 
{
    rn_08_11 = mach;
    
@if SH_VERSION == "1"
    if((rn_08_11 & 0x00000200) == 0) goto <ZERO_EXTEND>;

    # sign extend
    rn_08_11 = rn_08_11 | 0xFFFFFC00;
    goto <END>;

    <ZERO_EXTEND>
    rn_08_11 = rn_08_11 & 0x000003FF;
    
    <END>
@endif
}
</pre>

Reply:
Besides the case of instruction changes between versions, this seems like a non-issue as long as there are no collisions in the sleigh compiler.  If you're loading a sh-2, it shouldn't even have the bytes in the .text to decide into the newer instructions.  Or is there a case against this?

Keeping in mind this will likely be the core for sh-3 and sh-4, I agree with the file segments vs ifdef wrappers as that seems to scale better.  Also it may only be needed for collisions and handled case by case. 

My opinion is make this change when sh3/4 is added and the larger picture makes sense, but just get this initial support pulled.

Reply:
I spent some time looking at this. shad and shld are the only two SH-2A instructions I found that are not properly #ifdef-ed as SH-2A. As a result they were appearing in disassembly for SH-1/SH-2. I will take care of it shortly. 

> Besides the case of instruction changes between versions, this seems like a non-issue as long as there are no collisions in the sleigh compiler. If you're loading a sh-2, it shouldn't even have the bytes in the .text to decode into the newer instructions. Or is there a case against this?

This seems like a bad idea for a number of reasons. A processor module should only support the instructions it is supposed to. Off the top of my head you would break Ghidra's emulation feature.

Reply:
> My opinion is make this change when sh3/4 is added and the larger picture makes sense, but just get this initial support pulled.

Agreed. Let's keep the #ifdefs for now, assuming that all of them are correct.